presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java

```diff
@@ -150,7 +156,11 @@ private Range getFrameRange(FrameInfo frameInfo)
         int endPosition = partitionEnd - partitionStart - 1;
 
         // handle empty frame
-        if (emptyFrame(frameInfo, rowPosition, endPosition)) {
+        if (frameInfo.getType() == ROWS && emptyFrame(frameInfo, rowPosition, endPosition - rowPosition)) {
+            return new Range(-1, -1);
+        }
+
+        if (frameInfo.getType() == RANGE && emptyFrame(frameInfo, peerGroupStart, peerGroupEnd - 1)) {
```

Comment:
the last argument must be `endPosition - (peerGroupEnd - 1)` here, I will fix that in the next patch where we will have bounded `following`.